
# HG changeset patch
# User Tonya Souther <tonya.souther@gmail.com>
# Date 1350239636 18000
# Node ID 60e6ef631abb69d59de11b6244aedc7767fd1095
# Parent  c868e289e17a8174a841bfb3579ccbea364f3907
Allow colladadom library to be built standalone.

diff -r c868e289e17a -r 60e6ef631abb indra/cmake/ColladaDom.cmake
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/indra/cmake/ColladaDom.cmake	Sun Oct 14 13:33:56 2012 -0500
@@ -0,0 +1,15 @@
+# -*- cmake -*-
+include (Prebuilt)
+
+if (USESYSTEMLIBS)
+  find_path(COLLADA_INCLUDE_DIRS 1.4/dom/domConstants.h
+    PATH_SUFFIXES collada)
+  set(COLLADA_INCLUDE_DIRS
+    ${COLLADA_INCLUDE_DIRS} ${COLLADA_INCLUDE_DIRS}/1.4
+    )
+else (USESYSTEMLIBS)
+  set(COLLADA_INCLUDE_DIRS
+    ${LIBS_PREBUILT_DIR}/include/collada
+    ${LIBS_PREBUILT_DIR}/include/collada/1.4
+    )
+endif (USESYSTEMLIBS)
diff -r 827a910542a9 indra/llprimitive/CMakeLists.txt
--- a/indra/llprimitive/CMakeLists.txt	Mon Aug 17 14:54:41 2015 -0400
+++ b/indra/llprimitive/CMakeLists.txt	Sat Sep 19 12:13:18 2015 -0400
@@ -8,14 +8,16 @@
 include(LLXML)
 include(LLPhysicsExtensions)
 include(LLCharacter)
+include(ColladaDom)
 
 include_directories(
     ${LLCOMMON_INCLUDE_DIRS}
     ${LLMATH_INCLUDE_DIRS}
     ${LLMESSAGE_INCLUDE_DIRS}
     ${LLXML_INCLUDE_DIRS}
-    ${LIBS_PREBUILT_DIR}/include/collada
-    ${LIBS_PREBUILT_DIR}/include/collada/1.4
+    # ${LIBS_PREBUILT_DIR}/include/collada
+    # ${LIBS_PREBUILT_DIR}/include/collada/1.4
+    ${COLLADA_INCLUDE_DIRS}
     ${LLCHARACTER_INCLUDE_DIRS}
     )
 include_directories(SYSTEM
diff -r 827a910542a9 indra/newview/CMakeLists.txt
--- a/indra/newview/CMakeLists.txt	Mon Aug 17 14:54:41 2015 -0400
+++ b/indra/newview/CMakeLists.txt	Sat Sep 19 12:13:18 2015 -0400
@@ -7,6 +7,7 @@
 include(BuildPackagesInfo)
 include(BuildVersion)
 include(CMakeCopyIfDifferent)
+include(ColladaDom)
 include(DBusGlib)
 include(DirectX)
 include(DragDrop)
@@ -93,7 +94,7 @@
     ${LIBS_PREBUILT_DIR}/include/collada
     ${LIBS_PREBUILD_DIR}/include/hunspell
     ${OPENAL_LIB_INCLUDE_DIRS}
-    ${LIBS_PREBUILT_DIR}/include/collada/1.4
+    ${COLLADA_INCLUDE_DIRS}
     ${LLAPPEARANCE_INCLUDE_DIRS}
     ${CMAKE_CURRENT_SOURCE_DIR}
     )
